### ISSUE TITLE
`compile` should accept either file path or prebuilt sources object

### ISSUE THREAD

**Issue by d1ll0n** [huff-language/huffc#15]
The `compile` function takes a `HuffCompilerArgs` type as input.

```
type HuffCompilerArgs = {
  filePath: string;
  generateAbi: boolean;
  constructorArgs?: { type: string; value: string }[];
}
```

The `filePath` value is then passed to `parseFile` which passes it to `getAllFileContents`. `getAllFileContents` then reads the main file and all its imports and builds an object with the file contents of the main file and all its imports.

We should replace this with a more common pattern for compiler packages, where you can either provide an input file path or a prebuilt sources object containing all the file contents. This will make it possible to use huffc in environments without access to the filesystem, such as browsers.

**Comment by d1ll0n** [comment]
I had previously made a PR to the Aztec Protocol repository to enable this kind of behavior

https://github.com/AztecProtocol/huff/pull/5

There is some related discussion on that PR.

**Comment by d1ll0n** [comment]
Resolved in #16
